### PR TITLE
Delete password when private key is used

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

There is a bug with starting the debugger if the user once used a password, but is now using a private key, as the user would never be prompted for the correct password when debugging.

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/codefori/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

**Review by edmundreinhardt — APPROVED** [review]

File: src/webviews/settings/index.ts [hunk 1]
```diff
@@ -303,6 +303,7 @@ export class SettingsUI {
               if (!data.privateKeyPath?.trim()) {
                 if (connection.privateKeyPath?.trim()) {
                   data.privateKeyPath = connection.privateKeyPath;
+                  context.secrets.delete(`${name}_password`);
```
Comment (sebjulliand) [inline src/webviews/settings/index.ts:306]: It's an async call, we should await.
```suggestion
                  await context.secrets.delete(`${name}_password`);
```

**Review by sebjulliand — CHANGES_REQUESTED** [review]
It works as expected! Good job!
But `context.secrets` function are asynchronous, I'd suggest we `await` on them everywhere in the extension (here, but there are other places in the extension where `await` is missing too).

**Comment by worksofliam** [comment]
@sebjulliand Ready for another review - thanks!

**Review by sebjulliand — APPROVED** [review]
Looking good now!
I added a commit to add `await` in some other places where the secret store is used, I hope you won't mind 😁
Feel free to merge whenever you want.

**Pull request merged**